erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -72,6 +79,22 @@ def validate_stop_addresses(self):
 			if not stop.customer_address:
 				stop.customer_address = get_address_display(frappe.get_doc("Address", stop.address).as_dict())
 
+	def validate_delivery_note_not_draft(self):
+		delivery_notes = list(
+			set(stop.delivery_note for stop in self.delivery_stops if stop.delivery_note)
+		)
+		draft_delivery_notes = frappe.get_all(
+			"Delivery Note",
+			{"status": "Draft", "name": ["in", delivery_notes]},
```

Comment:
```suggestion
			{"docstatus": 0, "name": ["in", delivery_notes]},
```

The more reliable/standard way